modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -715,43 +768,53 @@ sub apply_tweaks {
                         Bio::EnsEMBL::Hive::Utils::PCL::parse_flow_into($self, $analysis, $new_value );
                     }
                 }
+                push @{$responseStructure->{Tweaks}}, $tweakStructure;
             }
 
         } elsif($tweak=~/^analysis\[([^\]]+)\]\.(\w+)(\?|#|=(.+))$/) {
+
             my ($analyses_pattern, $attrib_name, $operator, $new_value_str) = ($1, $2, $3, $4);
 
             my $analyses = $self->collection_of( 'Analysis' )->find_all_by_pattern( $analyses_pattern );
-            print "Tweak.Found   \t".scalar(@$analyses)." analyses matching the pattern '$analyses_pattern'\n";
+            push @response, "Tweak.Found   \t".scalar(@$analyses)." analyses matching the pattern '$analyses_pattern'\n";
 
             my $new_value = destringify( $new_value_str );
 
             foreach my $analysis (@$analyses) {
 
                 my $analysis_name = $analysis->logic_name;
-
+                my $tweakStructure;
+                $tweakStructure->{Object}->{Type} = 'Analysis';
+                $tweakStructure->{Object}->{Id} = $analysis->dbID + 0;
+                $tweakStructure->{Object}->{Name} = $analysis_name;
+                $tweakStructure->{Action} = $self->{ACTION}->{substr($operator, 0, 1)};
                 if( $attrib_name eq 'resource_class' ) {
+                    $tweakStructure->{Return}->{OldValue} = $analysis->resource_class ? $analysis->resource_class->name : undef;
+                    $tweakStructure->{Return}->{Field} = 'resource_class';
```

Comment:
Set `$tweakStructure->{Return}->{Field}` to `$attrib_name` before the `if`, which would cover all cases